src/main/webapp/app/entities/result/result.component.ts

```diff
@@ -100,7 +99,11 @@ export class ResultComponent implements OnInit, OnChanges, OnDestroy {
                 (this.participation.student && user.id === this.participation.student.id && (exercise.dueDate == null || exercise.dueDate.isAfter(moment()))) ||
                 (this.participation.student == null && this.accountService.isAtLeastInstructorInCourse(exercise.course))
             ) {
+                // unsubscribe old results if a subscription exists
                 // subscribe for new results (e.g. when a programming exercise was automatically tested)
+                if (this.websocketChannelResults) {
```

Comment:
So that we don't accidentally break something, the safest way is to unsubscribe before subscribing.
This way there can't be multiple listeners that all deliver results.
However I think there is a design issue in this component, as the only thing that could really change is the participation id and therefore the whole participation (and exercise).